eZ/Publish/Core/FieldType/Float/Type.php

```diff
@@ -233,6 +233,10 @@ protected function createValueFromInput( $inputValue )
      */
     protected function checkValueStructure( BaseValue $value )
     {
+        if ( is_int( $value->value ) )
+        {
+            $value->value = (float)$value->value;
+        }
         if ( !is_float( $value->value ) )
```

Comment:
i'd say no, cause $value could be a string or anything else and in the case exception needs to be thrown...